File: GolangSolutions/Array/M2575_找出字符串的可整除数组.go
/*
	@author Hyperspace
	@date 2024/03/07
	@file M2575_找出字符串的可整除数组.go
*/

package Array

func divisibilityArray(word string, m int) (res []int) {
	num := 0
	for _, w := range word {
		num = (num*10 + int(w-'0')) % m
		if num == 0 {
			res = append(res, 1)
		} else {
			res = append(res, 0)
		}
	}

	return
}

File: CPPSolutions/LeetCode/Array/M2575_找出字符串的可整除数组.cpp
#include <bits\stdc++.h>

using namespace std;

typedef long long LL;

class Solution {
public:
    vector<int> divisibilityArray(string word, int m) {
        vector<int> res;
        LL num = 0;
        for (auto& w : word) {
            num = (num * 10 + w - '0') % m;
            res.push_back(num == 0 ? 1 : 0);
        }

        return res;
    }
};
